keps/sig-node/4009-add-cdi-devices-to-device-plugin-api/README.md

```diff
@@ -0,0 +1,851 @@
+
+To a cluster admin upgrading to the newest API version, means upgrading Kubernetes to a newer version as well as upgrading the monitoring component.
+
+To a vendor changes in the API should always be backwards compatible.
+
+### Version Skew Strategy
+
+Kubelet will always be backwards compatible, so going forward existing plugins are not expected to break.
+<!--
+If applicable, how will the component handle version skew with other
+components? What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- Does this enhancement involve coordinating behavior in the control plane and
+  in the kubelet? How does an n-2 kubelet without this feature available behave
+  when this feature is used?
+- Will any other components on the node change? For example, changes to CSI,
+  CRI or CNI may require updating that component before the kubelet.
+-->
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+This change is not behind a feature gate. It is adding an optional
+field to a stable API. This mirrors what was done in []() when
+adding a `cpu_ids` field to the existing PodResources API.
+
+Furthermore, the only code that would be featured-gated in the Device Manager would be the code to propagate this field from the `AllocateResponse` to the
+corresponding field in the CRI.
```

Comment:
This change is not feature gated and no code is feature gated. I was calling out that *if a feature gate were to be added*, what would be guarded by it.

Reply:
For reference the Device Manager changes are: 
* https://github.com/kubernetes/kubernetes/pull/118254/files#diff-c07d343a8d459721656cf7ddacda71506bde01a5cac288d15a4aba6bfc9a1caeR676
* https://github.com/kubernetes/kubernetes/pull/118254/files#diff-26e686e6f2603e3f04deb5d3046314aaa966b787cfe23db979e6f4563b260c92R326-R339

Reply:
> This change is not feature gated and no code is feature gated. I was calling out that if a feature gate were to be added, what would be guarded by it.

I see. It's about how it would be done if a feature would be future gated, but it's not going to be future gated. Thank you for the explanations!

Reply:
Thanks @klueska. Removed and also added the link to https://github.com/kubernetes/enhancements/pull/1884 that was missing.